### DISCUSSION TITLE
Mapping of system tables and columns to custom ones

### DISCUSSION THREAD

**Discussion by manuelwehrli** [directus/directus#4900] [Feature Requests]
To use directus with a given data model (e.g. one that already has a user table), it would be perfect to have a mapping functionality to redirect certain system columns to custom columns, or whole tables to custom ones which have the required columns, allowing low coupling to proprietary directus tables without loosing core functionality like authentication, permissions, user management, etc. It could be done similar to the translation table mapping, or maybe more performant, reading it from a config file. The django framework might be an inspiration too, for example it solves custom user data model in a versatile way: https://docs.djangoproject.com/en/3.1/topics/auth/customizing/#specifying-a-custom-user-model

**Comment by benhaynes** [comment]
Love this idea. @WoLfulus has been asking for it forever. It's a big lift though, so let's get as much R&D here as we can — then we can dive in once we have the time/resources.

**Reply by benhaynes** [reply]
Looking into this further, for the App to work properly, there are a LOT of fields that would need to be mapped. Just looking at the **Directus Users** collection, the following are all used by the system. Also, it's worth noting that this "power-user" feature may not be worth adding a LOT of conditionals into the code/docs to dynamically disable the different features that require these fields...

```
- id
- email
- password
- first_name
- last_name
- avatar
- language
- theme
- 2FA
- status
- role
- token
```

While I like the concept here, I worry about the 80/20 of it given the added complexity. **Directus Files** might make more sense... but still, it's a heavy lift for something that likely won't be used by too many people.

**Reply by rijkvanzanten** [reply]
I'd rather focus our efforts on storing the directus stuff in a different data source first, and maybe look at decoupling files

**Reply by manuelwehrli** [reply]
The ability to map columns instead of only whole tables would avoid the problem of conditionals and disabling features. The idea is to still use directus system tables as a base, just proxying single columns to custom ones, for example first name, last name and email of a user, when they are already present in the custom data model. In the end let it be the responsability of the developer not to break directus and providing all the needed columns in a way. Can such a mechanism at least be considered to get its basics into the core before release 9, so a future extension or core feature can interact on that low level to provide such functionality without breaking changes?

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
